pkg/pillar/kubeapi/kubeapi.go

```diff
@@ -0,0 +1,283 @@
+// Copyright (c) 2024 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package kubeapi
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	netclientset "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/client/clientset/versioned"
+	"github.com/lf-edge/eve/pkg/pillar/base"
+	"github.com/lf-edge/eve/pkg/pillar/pubsub"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/retry"
+)
+
+const (
+	eveNameSpace        = types.EVEKubeNameSpace
+	kubeConfigFile      = types.EVEkubeConfigFile
+	vmiPodNamePrefix    = types.VMIPodNamePrefix
+	errorTime           = 3 * time.Minute
+	warningTime         = 40 * time.Second
+	stillRunningInerval = 25 * time.Second
+)
+
+// GetAppNameFromPodName : get application display name and also prefix of the UUID
+// from the pod name.
+func GetAppNameFromPodName(podName string) (displayName, uuidPrefix string, err error) {
```

Comment:
Note that I have done some changes in this file in my latest PR: https://github.com/naiming-zededa/eve/pull/246/files

Reply:
Also I'm working on preventing Kubernetes deps from being build for xen & kvm.
Maybe we should mark this PR as draft until I'm done (there will be small changes to this file).

Reply:
Yes I can mark it draft or I can close it and you can submit a new PR with these changes included. Whatever works for you.